### PR TITLE
164266827 transport operator view renewal

### PR DESCRIPTION
**Pull request by philosotractor**
# Changed
* Removed old redundant transport_operator implementation
* Renamed transport_operator_ytj to transport_operator
* ytj-integration within transport-operator, not outside
* removed ckan dependency on transport service side
* minor code optimisation

### CODE REVIEW

File: ote/src/cljs/ote/app/controller/transport_operator.cljs [hunk 1]
```diff
@@ -346,10 +337,10 @@
   CreateTransportOperator
   (process-event [_ app]
     (let [res (-> app
-                    (strip-ytj-metadata)
-                    (assoc
-                      :transport-operator {:new? true}
-                      :services-changed? true))]
+                  (strip-ytj-metadata)
+                  (assoc
+                    :transport-operator {:new? true}
```
Comment (rckopteri) [inline ote/src/cljs/ote/app/controller/transport_operator.cljs]: Could be on the same row as (assoc

File: ote/src/cljs/ote/app/controller/transport_operator.cljs [hunk 2]
```diff
@@ -386,16 +377,18 @@
...
-                           :transport-operator-loaded? true)
+                           :transport-operator-loaded? true
+                           :ytj-response {})
           nap-business-id (get-in state [:transport-operator ::t-operator/business-id])
           op-ytj-cache-miss? (or (empty? (:ytj-company-names state)) (not= nap-business-id (get-in state [:ytj-response :businessId])))]
-
-      (if op-ytj-cache-miss?
-        (send-fetch-ytj state nap-business-id)
-        (process-ytj-data state (:ytj-response state)))))
+      (if (flags/enabled? :open-ytj-integration)
```
Comment (rckopteri) [inline ote/src/cljs/ote/app/controller/transport_operator.cljs]: Should we turn the flag off from testing so we could make cypress tests work in CI and test server?

File: ote/src/cljs/ote/views/transport_operator.cljs [hunk 3]
```diff
@@ -8,131 +8,431 @@
...
+  (let [toggle-dialog (if (nil? (::t-operator/business-id operator))
+                        (to/->ToggleListTransportOperatorDeleteDialog operator)
+                        (to/->ToggleSingleTransportOperatorDeleteDialog))
+        operator-services (:transport-service-vector (some #(when (= (::t-operator/id operator) (get-in % [:transport-operator ::t-operator/id]))
+                                                                %)
+                                                           service-vector))]
   (when (:show-delete-dialog? operator)
     [ui/dialog
-     {:id "delete-transport-operator-dialog"
+     {:id (str "delete-transport-operator-dialog-" (::t-operator/id operator))
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator.cljs]: nice!

File: ote/src/cljs/ote/views/transport_operator.cljs [hunk 4]
```diff
@@ -8,131 +8,431 @@
...
+      ; label composition for existing business-id
+      (when (get-in state [:transport-operator :business-id-exists?])
+        {:element-id "label-business-id-is-not-unique"
+         :name :business-id-is-not-unique
+         :type :result-msg-warning
+         :content (tr [:common-texts :business-id-is-not-unique])}))))
+
+(defn- operator-form-groups [e! {operator :transport-operator :as state}]
+  "Creates a napote form and resolves data to fields. Assumes expired fields are already filtered from ytj-response."
+  ;(.debug js/console "operator-form-groups: state=" (clj->js state))
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator.cljs]: can be removed

File: ote/src/cljs/ote/views/transport_operator.cljs [hunk 5]
```diff
@@ -8,131 +8,431 @@
...
+  (let [response-ok? (= 200 (get-in state [:ytj-response :status]))
+        disable-ytj-address-billing? (= (get-in state [:ytj-flags :use-ytj-addr-billing?]) true)
+        disable-ytj-address-visiting? (= (get-in state [:ytj-flags :use-ytj-addr-visiting?]) true)
+        ytj-company-names (:ytj-company-names state)
+        ytj-company-names-found? (< 1 (count ytj-company-names))]
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator.cljs]: pos-int?

File: ote/src/cljs/ote/views/transport_operator.cljs [hunk 6]
```diff
@@ -8,131 +8,431 @@
...
+      (when (get-in state [:transport-operator :business-id-exists?])
+        {:element-id "label-business-id-is-not-unique"
+         :name :business-id-is-not-unique
+         :type :result-msg-warning
+         :content (tr [:common-texts :business-id-is-not-unique])}))))
+
+(defn- operator-form-groups [e! {operator :transport-operator :as state}]
+  "Creates a napote form and resolves data to fields. Assumes expired fields are already filtered from ytj-response."
+  ;(.debug js/console "operator-form-groups: state=" (clj->js state))
+  (let [response-ok? (= 200 (get-in state [:ytj-response :status]))
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator.cljs]: response-ok? would be easier to understand if with different name. Like ytj-response-ok? or something.

File: ote/src/cljs/ote/views/transport_operator.cljs [hunk 7]
```diff
@@ -8,131 +8,431 @@
...
+        {:name :msg-no-aux-names-for-business-id
+         :label (tr [:organization-page :no-aux-names-for-business-id])
+         :type :text-label
+         :max-length 128})
+
+      {:name       :msg-business-id-contact-details
+       :label      (if ytj-company-names-found?
+                     (tr [:organization-page :contact-details-plural])
+                     (tr [:organization-page :contact-details])
+                     )
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator.cljs]: trailing parentheses

**Comment by rckopteri** [comment]
Every pr that reduces code is good one :)

File: ote/src/cljs/ote/app/controller/transport_operator.cljs (hunk 2 above)
Comment (philosotractor) [reply ote/src/cljs/ote/app/controller/transport_operator.cljs]: Yes

**Comment by philosotractor** [comment]
Will implement some code formatting & optimisation improvements. Plus this needs a small change for flag-on mode.

File: ote/src/cljs/ote/views/transport_operator.cljs (hunk 6 above)
Comment (philosotractor) [reply ote/src/cljs/ote/views/transport_operator.cljs]: Yes

**Pull request merged**